content/src/content/jcr_root/apps/core/wcm/components/tableOfContent/v1/tableOfContent/tableOfContent.html

```diff
@@ -0,0 +1,26 @@
+<!--/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  ~ Copyright 2019 Adobe
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/-->
+<div class="cmp-tableOfContent"
+     data-sly-use.template="core/wcm/components/commons/v1/templates.html"
+     data-sly-use.tableOfContent="com.adobe.cq.wcm.core.components.models.TableOfContent"
+     data-sly-use.groupTemplate="group.html"
+>
+    <h1 class="cmp-tableOfContent__header">${tableOfContent.title}</h1>
+    <sly data-sly-call="${groupTemplate.group @ items=tableOfContent.items}">
+    </sly>
+
```

Comment:
Please add placeholder in case both `title` and `items` are empty. See https://github.com/adobe/aem-core-wcm-components/blob/master/content/src/content/jcr_root/apps/core/wcm/components/title/v2/title/title.html#L22

Reply:
To ease the work for front-end engineers who might want to customize the markup, we should keep the markup as simple as possible, and avoid any logic that can be done by the model. I'd argue that the model should provide a clean list of items that HTL doesn't need to test for empty items (other than the overall list being empty and the placeholder needs to be displayed).